core/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoHashGridAggregator.java

```diff
@@ -66,6 +70,28 @@ public boolean needsScores() {
     }
 
     @Override
+    public void collectBucket(LeafBucketCollector subCollector, int doc, long bucketOrd) throws IOException {
+        bucketCentroids = bigArrays.grow(bucketCentroids, bucketOrd + 1);
+        super.collectBucket(subCollector, doc, bucketOrd);
+    }
+
+    protected final void adjustCentroid(long bucketOrd, long geohash) {
+        final int numDocs = getDocCounts().get(bucketOrd);
+        final GeoPoint oldCentroid = new GeoPoint();
+        final GeoPoint nextLoc = new GeoPoint();
+
+        if (numDocs > 1) {
+            final long curCentroid = bucketCentroids.get(bucketOrd);
+            oldCentroid.resetFromGeoHash(curCentroid);
```

Comment:
is it worth doing the conversion from and to geohash every time here? Could it be better to not do the conversion and store two doubles per bucket instead of one long? I guess its a trade-off between time and memory